src/NetEscapades.EnumGenerators/EnumGenerator.cs

```diff
@@ -222,7 +221,7 @@ static void Execute(in EnumToGenerate enumToGenerate, SourceProductionContext co
                 isDisplayNameTheFirstPresence = displayNames.Add(displayName);    
             }
             
-            members.Add((member.Name, new EnumValueOption(displayName, isDisplayNameTheFirstPresence)));
+            members.Add((member.Name, new EnumValueOption(displayName, isDisplayNameTheFirstPresence, Convert.ToDecimal(constantValue))));
```

Comment:
I've just understood what you were talking about about `object` in your other comment 🤦 i.e. it's _already_ boxed here 🙄 So yeah, what you had before probably _did_ make the most sense, in that there were no additional allocations, sorry 😬

Reply:
I actually checked again, if there was another way to get that value. Bonus: you already retrieved the boxed value before, so even, if it is only boxed on access, the boxing already happened before